vector-search-index.md

```diff
@@ -0,0 +1,263 @@
+- Directly dropping columns with vector search indexes is not supported. You can drop such a column by first dropping the vector search index on that column and then dropping the column itself.
+- Modifying the type of a column with a vector index is not supported.
+- Setting vector search indexes as [invisible](/sql-statements/sql-statement-alter-index.md) is not supported.
+- Building vector search indexes on TiFlash nodes with [encryption at rest](https://docs.pingcap.com/tidb/stable/encryption-at-rest) enabled is not supported.
+
+## Create the HNSW vector index
+
+[HNSW](https://en.wikipedia.org/wiki/Hierarchical_navigable_small_world) is one of the most popular vector indexing algorithms. The HNSW index provides good performance with relatively high accuracy, up to 98% in specific cases.
+
+In TiDB, you can create an HNSW index for a column with a [vector data type](/vector-search-data-types.md) in either of the following ways:
+
+- When creating a table, use the following syntax to specify the vector column for the HNSW index:
+
+    ```sql
+    CREATE TABLE foo (
+        id       INT PRIMARY KEY,
+        embedding     VECTOR(5),
+        VECTOR INDEX idx_embedding ((VEC_COSINE_DISTANCE(embedding)))
+    );
+    ```
+
+- For an existing table that already contains a vector column, use the following syntax to create an HNSW index for the vector column:
+
+    ```sql
+    CREATE VECTOR INDEX idx_embedding ON foo ((VEC_COSINE_DISTANCE(embedding))) USING HNSW;
+    ALTER TABLE foo ADD VECTOR INDEX idx_embedding ((VEC_COSINE_DISTANCE(embedding))) USING HNSW;
+
+    -- You can also explicitly specify "USING HNSW" to build the vector search index.
+    CREATE VECTOR INDEX idx_embedding ON foo ((VEC_COSINE_DISTANCE(embedding))) USING HNSW;
+    ALTER TABLE foo ADD VECTOR INDEX idx_embedding ((VEC_COSINE_DISTANCE(embedding))) USING HNSW;
+    ```
+
+> **Note:**
+>
+> The vector search index feature relies on TiFlash replicas for tables.
+>
+> - If a vector search index is defined when a table is created, TiDB automatically creates a TiFlash replica for the table.
+> - If no vector search index is defined when a table is created, and the table currently does not have a TiFlash replica, you need to manually create a TiFlash replica before adding a vector search index to the table. For example: `ALTER TABLE 'table_name' SET TIFLASH REPLICA 1;`.
+
+When creating an HNSW vector index, you need to specify the distance function for the vector:
+
+- Cosine Distance: `((VEC_COSINE_DISTANCE(embedding)))`
+- L2 Distance: `((VEC_L2_DISTANCE(embedding)))`
+
+The vector index can only be created for fixed-dimensional vector columns, such as a column defined as `VECTOR(3)`. It cannot be created for mixed-dimensional vector columns (such as a column defined as `VECTOR`) because vector distances can only be calculated between vectors with the same dimension.
```

Comment:
```suggestion
The vector index can only be created for fixed-dimensional vector columns, such as a column defined as `VECTOR(3)`. It cannot be created for non-fixed-dimensional vector columns (such as a column defined as `VECTOR`) because vector distances can only be calculated between vectors with the same dimension.
```